pkg/monitor/ifindex.go

```diff
@@ -6,11 +6,23 @@ package monitor
 import (
 	"fmt"
 
-	"github.com/cilium/cilium/pkg/datapath/link"
+	"github.com/cilium/cilium/pkg/hubble/parser/getters"
 )
 
+// linkMonitor watches for links on the system so that ifindexes can be
+// string-formatted with human-readable interface names rather than the
+// integer values.
+//
+// Package-level variable just to avoid overhauling old code to introduce an
+// overarching structure to help format the messages nicely.
+var linkMonitor getters.LinkGetter
```

Comment:
I'm not the biggest fan of this, but at least it's abstracted so there's no direct dependency from pkg/monitor onto the Linux kernel. Unlike the `pkg/hubble` code, `pkg/monitor` doesn't seem to have the same notion of a parser structure that could hold the pointer to this (from the daemon), so I just manually plumbed a package-level variable here instead as baby steps towards cleaner code.